setup.py

```diff
@@ -50,6 +74,9 @@
         'ephem',
         'six',
         'pyserial',  # undeclared dependency of twisted.internet.serialport
+        'service_identity',
```

Comment:
Each of these added dependencies should have a comment explaining why they are needed despite not being imported, like the entry for pyserial above.

Reply:
I see you added comments, but they do not explain the dependencies. What is "urllib3" being used by? This script itself, which imports "urllib"? If so, say so. Also, service_identity is still unexplained.

Sorry to be so nitpicky about this, but it would be bad to have code that is both not commented sufficiently and which _I_ don't fully understand. Therefore I insist on sufficient explanations in the comments here.